cmd/libs/go2idl/deepcopy-gen/generators/deepcopy.go

```diff
@@ -384,8 +385,12 @@ func (g *genDeepCopy) Init(c *generator.Context, w io.Writer) error {
 	sw.Do("if err := $.scheme|raw$.AddGeneratedDeepCopyFuncs(\n", map[string]interface{}{
 		"scheme": scheme,
 	})
+	reflect := c.Universe.Type(types.Name{Package: "reflect", Name: "TypeOf"})
+	runtime := c.Universe.Type(types.Name{Package: runtimePackagePath, Name: "GeneratedDeepCopyFunc"})
 	for _, t := range g.typesForInit {
-		sw.Do(fmt.Sprintf("%s,\n", g.funcNameTmpl(t)), argsFromType(t))
+		g.imports.AddType(reflect)
+		g.imports.AddType(runtime)
+		sw.Do(fmt.Sprintf("runtime.GeneratedDeepCopyFunc{Fn: %s, InType: reflect.TypeOf(func() *$.type|raw$ {var x *$.type|raw$; return x}())},\n", g.funcNameTmpl(t)), argsFromType(t))
```

Comment:
I've got 2 concerns.  First, I was trying to decouple deepcopy generation from deep kubernetes packages, this moves in the wrong direction for that.  Second, calling by name doesn't require that all functions are registered, and there are many that are not.  I am concerned that there will be cases that call-by-name linked and lookup-regsitration will fail.  Can you convince me?